Automattic/jetpack
# Add logic to fetch intro offers

#### Changes proposed in this Pull Request:

This PR adds the logic to fetch the Jetpack introductory offers. In practice, we'll instantiate the data component `<QueryIntroOffers>` to fetch the offers.

#### Jetpack product discussion


Required to implement the timed discount. Context: pbNhbs-1xX-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






- Pull the branch
- Update `projects/plugins/jetpack/_inc/client/recommendations/index.jsx` to instantiate `<QueryIntroOffers>`, as shown below
  <img width="992" alt="Screen Shot 2022-04-07 at 11 17 20 AM" src="https://user-images.githubusercontent.com/1620183/162234661-6f8fb071-2cf1-4485-8056-2a7e5938b996.png">
- Rebuild the plugin
- Open your browser Network panel
- Launch your test site and visit `/wp-admin/admin.php?page=jetpack#/recommendations/site-type`
- Check in the Network panel that the endpoint `/wp-json/jetpack/v4/intro-offers` was called, and that it successfully returned the intro offers data
  <img width="628" alt="Screen Shot 2022-04-07 at 11 08 29 AM" src="https://user-images.githubusercontent.com/1620183/162234895-6282f7d8-5e2e-49f9-8097-3e99b2a5f46c.png">
